enterprise/backup/src/main/java/org/neo4j/backup/nextgen/BackupModule.java

```diff
@@ -0,0 +1,146 @@
+        pageCache = aPageCache( fileSystemAbstraction );
+        clock = Clock.systemDefaultZone();
+        monitors = new Monitors();
+        sslPolicy = sslPolicy( config, logProvider );
+        catchUpClient = catchUpClient();
+        storeCopyClient = new StoreCopyClient( catchUpClient, logProvider );
+        backupCommandArgumentHandler = new BackupCommandArgumentHandler();
+    }
+
+    public BackupDelegator backupDelegator( File storeDir, AdvertisedSocketAddress fromAddress )
+    {
+        return backupDelegator( storeCopyService( storeDir, fromAddress ) );
+    }
+
+    public FileSystemAbstraction fileSystemAbstraction()
+    {
+        return fileSystemAbstraction;
+    }
+
+    public BackupCommandArgumentHandler backupCommandArgumentHandler()
+    {
+        return backupCommandArgumentHandler;
+    }
+
+    private BackupDelegator backupDelegator( StoreCopyService storeCopyService )
+    {
+        return new BackupDelegator( remoteStore(), catchUpClient, storeCopyClient, storeCopyService,
+                ExternallyManagedPageCache.graphDatabaseFactoryWithPageCache( pageCache ), new ClearIdService( new IdGeneratorWrapper() ) );
+    }
+
+    private StoreCopyService storeCopyService( File storeDir, AdvertisedSocketAddress fromAddress )
+    {
+        return new StoreCopyService( storeDir, fileSystemAbstraction, pageCache, storeCopyClient, fromAddress );
+    }
+
+    private RemoteStore remoteStore()
+    {
+        Monitors monitors = new Monitors();
+        TxPullClient txPullClient = new TxPullClient( catchUpClient, monitors );
+        TransactionLogCatchUpFactory txLogFactory = new TransactionLogCatchUpFactory();
+        LogProvider logProvider = NullLogProvider.getInstance();
+        return new RemoteStore( logProvider, fileSystemAbstraction, pageCache, storeCopyClient, txPullClient, txLogFactory, monitors );
+    }
+
+    private CatchUpClient catchUpClient()
+    {
+        return new CatchUpClient( logProvider, clock, inactivityTimeoutMillis, monitors, sslPolicy );
+    }
+
+    private static SslPolicy sslPolicy( Config config, LogProvider logProvider )
+    {
+        SslPolicyLoader sslPolicyLoader = SslPolicyLoader.create( config, logProvider );
+        return sslPolicyLoader.getPolicy( config.get( CausalClusteringSettings.ssl_policy ) );
+    }
+
+    private static PageCache aPageCache( FileSystemAbstraction fileSystemAbstraction )
+    {
+        PageCursorTracerSupplier pageCursorTracerSupplier = new DefaultPageCursorTracerSupplier();
+        PageCacheTracer pageCacheTracer = new DefaultPageCacheTracer();
+        int cachePageSize = 8192;
+        int maxPages = 1024;
+        PageSwapperFactory pageSwapperFactory = new SingleFilePageSwapperFactory();
+        pageSwapperFactory.setFileSystemAbstraction( fileSystemAbstraction );
+        return new MuninnPageCache( pageSwapperFactory, maxPages, cachePageSize, pageCacheTracer, pageCursorTracerSupplier );
```

Comment:
Are you intentionally ignoring the page cache configurations in `neo4j.conf`? In doing so, we can no longer do an online backup directly to block device storage, but have to change the procedure to back up to file storage and then import onto block device storage.